libvmi/driver/xen/xen_events_abi.h

```diff
@@ -472,6 +477,7 @@ typedef struct vm_event_st_412 {
         struct vm_event_mov_to_msr_411        mov_to_msr;
         struct vm_event_desc_access           desc_access;
         struct vm_event_singlestep            singlestep;
+        struct vm_event_fast_singlestep       fast_singlestep;
```

Comment:
You won't be able to add this to `vm_event_st_412`, or even `vm_event_st_413`. This feature will likely only get merged into Xen 4.14, so you'll need to base your changes on the current xen-unstable branch. In there, the vm_event structure has already changed, so we'll need to implement a new `vm_event_st_414` here. I usually wait with such changes to add support in LibVMI for until the Xen RC's are coming since there could be other changes made to the vm_event structure before the merge window is closed.

Reply:
May be we could back-port the feature into https://github.com/tklengyel/xen ?

Reply:
You can backport it to any version, but as long as its not upstream I wouldn't merge it to LibVMI master. So if you are using with Xen 4.12 I suggest you have a fork for that.

Reply:
OK. I will fix it in a while.